lib/rspec-puppet/support.rb

```diff
@@ -15,6 +15,27 @@ def environment
       'rp_env'
     end
 
+    def stub_file_consts(example)
+      munged_facts = facts_hash(nodename(example.metadata[:type]))
+
+      if munged_facts['operatingsystem'] && munged_facts['operatingsystem'].to_s.downcase == 'windows'
+        stub_const_wrapper('PATH_SEPARATOR', ';')
+        stub_const_wrapper('ALT_SEPARATOR', "\\")
+      else
+        stub_const_wrapper('PATH_SEPARATOR', ':')
+        stub_const_wrapper('ALT_SEPARATOR', nil)
```

Comment:
Seeing those values hardcoded here makes me very itchy. Please add a set of tests to verify the underlying ruby platform like this:

```
# make sure we do not miss changes in the ruby platform relevant for lib/rspec-puppet/support.rb stub_file_consts
describe "consts on windows", :if => RSpec::Support::OS.windows? do
  it { expect(File.PATH_SEPARATOR).to eq(':') }
  it { expect(File.ALT_SEPARATOR).to be_nil }
end

describe "consts on linux", :unless => RSpec::Support::OS.windows? do
  it { expect(File.PATH_SEPARATOR).to eq(';') }
  it { expect(File.ALT_SEPARATOR).to eq("\\") }
end
```